### PR TITLE
mzcloud-cli

### PR DESCRIPTION
**Pull request by alex-hunt-materialize**

### CODE REVIEW

File: src/mzcloud-cli/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+use mzcloud::apis::{
+    configuration::Configuration,
+    deployments_api::{
+        deployments_certs_retrieve, deployments_create, deployments_destroy, deployments_list,
+        deployments_logs_retrieve, deployments_retrieve, deployments_update,
+    },
+    mz_versions_api::mz_versions_list,
+    schema_api::schema_retrieve,
+};
+use mzcloud::models::{deployment_request::DeploymentRequest, deployment_size::DeploymentSize};
```
Comment (benesch) [inline src/mzcloud-cli/src/main.rs:27]: Can you configure your rust analyzer import style to "last"? That's the style we've weakly settled on.

**Review by benesch — COMMENTED** [review]
🎉 This is so exciting! A few quick high level thoughts:

  * @rjnn proposes that we call the binary just "mzcloud" rather than "mzcloud-cli", and I agree! I think that's as simple as moving src/main.rs to src/bin/mzcloud.rs. Leaving the crate name as "mzcloud-cli" is probably right.
  * We probably want to introduce a layer of resource nesting right from the get go? Like, `mzcloud deployment {create,delete,list}`, to make space for `mzcloud tls-authority {create,delete,list}` down the road?

File: src/mzcloud-cli/src/main.rs (hunk 1 above)
Comment (alex-hunt-materialize) [reply src/mzcloud-cli/src/main.rs:27]: Do you mean `importGranularity = "module"`?

Comment (benesch) [reply src/mzcloud-cli/src/main.rs:27]: Oh, I guess so. Seems like they renamed that option.

**Comment by benesch** [comment]
Just to jot down a few things from our live chat today:

  * This broadly looks awesome!
  * Let's plan to ditch the `schema` command for now since it's easy to add back later.
  * `mzcloud mz-versions list` seems fine for now, but might ultimately change.
  * We should teach the backend API not to require the `mz_version` field, and it can default to the latest if unspecified.

**Comment by benesch** [comment]
Thanks again for this, Alex! I made a number of minor changes that I'll try to summarize here.

  * Omitting the version field defaults to the latest version. So I changed the CLI to skip discovering the latest version when creating a 
  * Partial updates work now, so long as you use `PATCH` and not `PUT`, which I changed the CLI to do.
  * I pluralized the `deployment` subcommand to `deployments` to match the `mz-version` command.
  * I adjusted the Frontegg domain computation to use the vanity materialize.com domain (`admin.[staging.]cloud.materialize.com`), so that you don't have to separately manage the OAuth domain.
  * I tweaked a few imports and names (`Opts` => `Args`, `std::fs` => `use std::fs; fs`) to match the style we use elsewhere.
  * I added a Docker image configuration to build and publish mzcloud to Docker Hub as `materialize/mzcloud`. That should solve the immediate distribution needs, though I'd love to get this into the Debian and Homebrew packages too.

I just filed #7860 to track the various features I've thought of. But going to merge for now, to kick things off!

**Review by benesch — APPROVED** [review]

**Pull request merged**